### DISCUSSION TITLE
Renaming rg3d!

### DISCUSSION THREAD

**Discussion by gideongrinberg** [FyroxEngine/Fyrox#103] [General]
We'd been discussing renaming rg3d on the Discord, but I thought it would be a good idea to have a discussion thread about it as well.

**Comment by ghost** [comment]
https://pigeonhole.at/RG3D -  here's the poll for

**Comment by ghost** [comment]
So community came to conclusion to not rename the engine.

**Comment by gideongrinberg** [comment]
The community didn't come to any conclusion. You ran a random poll ;p

**Comment by timleg002** [comment]
Hey! I would also be up for a name change. However, I do respect if the author wishes that the engine has the original name the author intended. But, it's not really a cool name ;p and I may have some prejudices against it with this name. But all in all, it's a very cool project and definitely would use it!

**Comment by mrDIMAS** [comment]
I decided to rename the engine. I dislike current name.

My options are:

- [Pyroxen](https://en.wikipedia.org/wiki/Pyroxene)
- Pyrox (shorter version of Pyroxen) 
- Cryxen (**Cry**ptos + **xen**os)
- Feox (**Fe**rrum + **ox**ide) - my favorite for now

@adamnemecek proposed following options:

- Pulse
- Gamma ray
- Peroxide
- Spectral
- Specter 
- Xenon
- Phantasmal
- Orbital

I'm open for any other options.

**Reply by romancerny** [reply]
I personally like rg3d, but I'm not the main person to decide

**Reply by gogo2464** [reply]
I like rg3d too but if we plan to make more 2D, it will be embarassing.

**Reply by romancerny** [reply]
I had an impression that 2D is the subset of 3D

**Reply by alice-i-cecile** [reply]
Pyroxen is a wonderful name: distinctive, interesting, easy to pronounce.

Pyrox is similar, but not quite as nice to read.

Cryxen is solid.

Feox has nice connections to Rust, but I worry that you'll struggle to find your own symbolism for branding. It's also quite hard to figure out how to pronounce. Ferox definitely sounds better.

**Comment by ValorZard** [comment]
Bringing in Ferox from the discord, since its like Feox but sounds more "right"

**Comment by dawnlarsson** [comment]
SilverAlloy - A un-used engine brand (logo and all) i've had on my backburner for the last while, would not mind if it found a better place. :)

**Comment by Eschryn** [comment]
Limn, Limn Engine

> Definition of limn
transitive verb
1: to draw or paint on a surface

https://www.merriam-webster.com/dictionary/limn

In Terms of syllables, this is the shortest game engine name I can think of - I don't know any other game engine name where the name is just one syllable. Quite unique

**Comment by parhamgholami** [comment]
I didn't come up with these names myself; however, based on the Discord conversation, I'm partial to:

Alloy
Rustica
Dawn

**Comment by toyboot4e** [comment]
Carci Engine, which brings us [carcinisation][C] to be 3D crabs

[C]: https://en.m.wikipedia.org/wiki/Carcinisation

**Comment by timleg002** [comment]
I suggest "Ray"

**Comment by martin-t** [comment]
I wanted to suggest AarGee / Aargee / Argee. It's alphabetically first of all current engines and maintains some continuity with rg3d. However, there are already many google hits for these words.

That being said, bevy was also already taken and the engine is currently not even the first hit. Marketing it as **AarGee Engine** might work well enough.

Feox is a really good choice - i don't see any domains with such name. DDG currently thinks it's misspelled "fox" and displays fox news but i think that should fix itself once you get a domain with Feox in it.

Finally, i wanna say that i think it should not be a common word like ray, alloy, pulse, etc. It can be an existing english word but should be uncommon enough to stand out. Also wanna advise against Cryxen because of the negative association with the word cry.

**Comment by dmitrykolesnikovich** [comment]
Just http://just.rs

**Comment by parhamgholami** [comment]
Another one that might be worth considering:

Ludo or Ludere

Ludo is Latin for "I play" and Ludere more generally translates to "play".

**Comment by mrDIMAS** [comment]
Here's the list of 5 names of what I find as good candidates, this list constructed from every option that I saw here and on Discord server:

- Ferox
- Limn - (there is a GUI library with the same name)
- Pyroxen
- Liminal
- Molten

**Reply by romancerny** [reply]
I like Liminal the most

**Reply by Barugon** [reply]
There's already a [Molten](https://moltengl.com/moltenvk/) library.

**Comment by dmitrykolesnikovich** [comment]
http://contrust.info

**Comment by Barugon** [comment]
If you decide to keep it rg3d then consider moving "engine" to the end; i.e. Rust Game 3D engine or rg3d engine.

**Comment by Barugon** [comment]
Or, why not just drop the 3D part and call it Rust Game Engine?

**Reply by jexxadox** [reply]
well. that would claim its officially part of the core rust program creators wouldn't it?

**Comment by jexxadox** [comment]
as a newcomber my thoughts: wasn't a name. was a list of numbers and letters and literally tuned it out....  but im also new to github and the massive amount of text on the main pages still freaks me out lol....
was looking for an "unreal UDK" version using rust. i think i found it !

**Comment by Barugon** [comment]
Okay, why Fyrox? I suppose it's as good as anything else; just curious about the reasoning.

[edit] I'm also trying to grok the logo.

**Reply by mrDIMAS** [reply]
It is originating from initial `Feox` name that I proposed back in the day, people told me that it sound like it missing a letter `r` which turned the name into `Ferox`, but then we found that `Ferox` was already taken and then I decided to change `e` to `y` and the `Fyrox` was born. The logo was inspired by the `Fyr = fire` + `ox = bull`, it gave the fire bull logo :)

**Reply by Barugon** [reply]
Thanks for the info.

**Comment by Barugon** [comment]
Had you considered Hematite (essentially rust in crystal form)?

**Reply by Barugon** [reply]
Or Magnetite (higher order iron oxide crystal)?

**Reply by ghost** [reply]
there was a voting, and Fyrox won

**Reply by ghost** [reply]
Hematite was in the voting too

**Comment by felipetesc** [comment]
Rave Game Engine, or Rave Engine
Blast Game Engine - Blast Engine
Eruption  Game Engine - Eruption 
Storm Game Engine - 
Tempest Game Engine
Tsunami Game Engine
R Game Engine, or Are Game Engine
Supernova Game Engine
Hurricane Game Engine
Rune Game Engine
Spelter (Another name for Zinc)
Californium  Game Engine
Fermium  Game Engine
Check the url for a list of "synthetic metallic elements"
https://en.wikipedia.org/wiki/Synthetic_element#:~:text=All%20elements%20with%20atomic%20numbers,before%20being%20found%20in%20nature.
